### DISCUSSION TITLE
Test RAM cells

### DISCUSSION THREAD

**Discussion by Foxite** [hneemann/Digital#1207] [Q&A]
When using test vectors to load and run a test program, I want to test that the contents of RAM match what I specified.

In my case I have a program that fills every RAM cell with a value equal to its address. I want to run this program and test that this has been done.

What's the best way to test RAM from test vectors?

**Comment by Foxite** [comment]
For now I have solved this by adding a secondary RAM component (dual port) that receives the same writes as the main RAM component, but its second port is controlled by pins used by test vectors.

Though it would be nice if I could peek specific RAM addresses from test vectors

**Comment by hneemann** [comment]
In the file `examples/processor/ProcessorTest.dig` you can find all test cases created for my processor.
I did it by writing a register into a memory location and then reading the value into another register.
